backend/tracim_backend/views/core_api/user_controller.py

```diff
@@ -162,6 +163,18 @@
 ]
 
 
+def has_write_rights(field: str, app_config: CFG, request: TracimRequest) -> bool:
+    """
+    Check if user can write provided field (is admin or field is writable)
+    """
+    read_only_fields = app_config.USER__READ_ONLY_FIELDS.get(request.candidate_user.auth_type.value)
+    if request.current_user.profile.name != "ADMIN":
```

Comment:
- I suggest not to allow admin to edit readonly fields
- If we allow (let me know if you see a very good reason for this), you *must not* compare with a string `ADMIN`. In this case, you should use `request.current_user.profile != Profile.ADMIN:`

Strings are evil.

Reply:
Ok I see. Indeed, my bad for matching with strings and not profile itself.

As for the admins, I'll let @PhilippeAccorsi have a word about this since he was the first to bring this subject.

Reply:
@Millefeuille42 there are two types of administrators:

- technical administrators who are in charge of runtime. They access to CLI and have the spirit of "not touching things that may break stuff".
- functionnal administrators. The are in charge of functionnal parameterization of softwares. The use a UI and consider that software can't be break.

Most of the time, things available through graphical UI should not break stuff while CLI opeartions may break stuff.

@PhilippeAccorsi what do you think?

Reply:
I get your point. I do agree then. I'll still let Philippe answer before doing anything else

Reply:
@Millefeuille42 @lebouquetin

Reply:
Read only status is only config wise (not "set", it is evaluated depending on config). It still should be editable through SQL and CLI. Additionally, the read-only status is not considered if data is changed on the external system.

Reply:
removed rights for admin to edit read-only fields in last commit